WordPress/Classes/Services/ReaderTopicService.m

```diff
@@ -550,11 +550,11 @@ - (void)siteTopicForSiteWithID:(NSNumber *)siteID
 {
     ReaderSiteTopic * __block siteTopic = nil;
 
-    [self.coreDataStack performAndSaveUsingBlock:^(NSManagedObjectContext *context) {
+    [self.coreDataStack.mainContext performBlockAndWait:^{
         if (isFeed) {
-            siteTopic = [ReaderSiteTopic lookupWithFeedID:siteID inContext:context];
+            siteTopic = [ReaderSiteTopic lookupWithFeedID:siteID inContext:self.coreDataStack.mainContext];
         } else {
-            siteTopic = [ReaderSiteTopic lookupWithSiteID:siteID inContext:context];
+            siteTopic = [ReaderSiteTopic lookupWithSiteID:siteID inContext:self.coreDataStack.mainContext];
```

Comment:
Out of topic, and apologies if this has already been discussed. 

Have you ever considered passing the context to the block in `performBlockAndWait`? I had a look at the first dozen results from searching the codebase for `performBlockAndWait` and most of them called the method on `someContext` and then accessed `someContext` in the block. That's not wrong and I don't think it would lead to runtime issues, but it seems to me it would be more convenient to access the context as a block parameter

```objc
[self.coreDataStack.mainContext performBlockAndWait:^(NSManagedObjectContext *context) {
    siteTopic = [ReaderSiteTopic lookupWithFeedID:siteID inContext:context];
}];

Reply:
I have had the same thought many times! But, it's up to Apple I guess, since [it's a Core Data API](https://developer.apple.com/documentation/coredata/nsmanagedobjectcontext/1506364-performblockandwait).

However, I did add a [`performQuery`](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Classes/Utility/CoreDataHelper.swift#L186) Swift-only function to do what you suggested. I guess we could have a similar API for Objective-C.

Reply:
> But, it's up to Apple I guess, since [it's a Core Data API](https://developer.apple.com/documentation/coredata/nsmanagedobjectcontext/1506364-performblockandwait).

🤦‍♂️ 

I've been looking at our `perform*` methods so much that I got totally blindsided by this one being an Apple one.